### PR TITLE
Another take on projectionist config

### PR DESCRIPTION
**Pull request by kelvinst**
This is just to incremento your PR https://github.com/elixir-editors/vim-elixir/pull/451 with the stuff @hauleth commented + some dispatch config.

Also, I changed it so it to change the `heuristics` option instead of adding the projections on the existent ones, because the way you had it before was not possible to use `:Pcd` to change directories to the closest `mix.exs` file.

Thanks!

### CODE REVIEW

File: ftplugin/elixir.vim [hunk 1]
```diff
@@ -52,18 +52,52 @@ silent! setlocal formatoptions-=t formatoptions+=croqlj
...
+        \     },
+        \     '*.exs': {
+        \       'makery': {
+        \         'lint': { 'compiler': 'credo' },
+        \         'test': { 'compiler': 'exunit' },
+        \         'build': { 'compiler': 'mix' }
+        \       }
+        \     }
+        \   }
+        \ })
```
Comment (hauleth) [inline ftplugin/elixir.vim]: I would use 3rd argument with value `keep` to not overwrite user-defined projections.

File: ftplugin/elixir.vim [hunk 2]
```diff
@@ -52,18 +52,52 @@ silent! setlocal formatoptions-=t formatoptions+=croqlj
...
+        \     },
+        \     'test/*_test.exs': {
+        \       'type': 'test',
+        \       'alternate': 'lib/{}.ex',
+        \       'dispatch': "mix test test/{}_test.exs`=v:lnum ? ':'.v:lnum : ''`",
+        \       'template': [
+        \         'defmodule {camelcase|capitalize|dot}Test do',
+        \         '  use ExUnit.Case',
+        \         '',
+        \         '  alias {camelcase|capitalize|dot}, as: Subject',
```
Comment (hauleth) [inline ftplugin/elixir.vim]: I do not think that this line is common practice. Even I am questioning it currently.

**Review by hauleth — CHANGES_REQUESTED** [review]

File: ftplugin/elixir.vim (hunk 1 above)
Comment (kelvinst) [reply ftplugin/elixir.vim]: Yep, doing it

File: ftplugin/elixir.vim (hunk 2 above)
Comment (kelvinst) [reply ftplugin/elixir.vim]: Ture, I will remove the alias and the reference to it.

**Comment by kelvinst** [comment]
Closing it as I just don't use my projectionist config like that anymore.

**Pull request closed without merging**